client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,352 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
   return results;
 };
 
-const getResultsPreview = (searchTerm: string, state, params: SearchParams) => {
-  let results;
-  let types;
-  const { communityScope, chainScope, isHomepageSearch } = params;
-  if (communityScope || chainScope) {
-    types = [SearchType.Discussion, SearchType.Member];
-    results = getBalancedContentListing(app.searchCache[searchTerm], types);
-  } else if (isHomepageSearch) {
-    types = [SearchType.Community];
-    results = getBalancedContentListing(app.searchCache[searchTerm], types);
-  } else {
-    types = [SearchType.Discussion, SearchType.Member, SearchType.Community];
-    results = getBalancedContentListing(app.searchCache[searchTerm], types);
-  }
+const getResultsPreview = (searchQuery: SearchQuery, state) => {
+  const { communityScope } = searchQuery;
+  const types = searchQuery.getSearchScope()
+  const results = getBalancedContentListing(app.search.getByQuery(searchQuery).results, types)
```

Comment:
Getting an error thrown here that `.results` is being called on null, try `?.`

A handful of errors are throwing whenever I try to go directly to/hard-refresh a full Search page. Ostensibly, they have to do with the initialization not being finished yet? Not sure.

Try directly loading/hard-refreshing something like `http://localhost:8080/search?q=the&cutting&sort=Best` and clear up any errors you get?

Reply:
This was caused by a bug with the way I was encoding the url - the search term for that url was only being read as "the" rather than "the cutting". I added an error page to handle invalid searchs on the page and fixed the url bug - you can try `http://localhost:8080/search?q=the%20cutting&sort=Best` to verify